demos/device_defender_for_aws/report_builder.c

```diff
@@ -482,9 +641,48 @@ ReportBuilderStatus_t GenerateJsonReport( char * pBuffer,
         }
     }
 
+    /* Write task ids array. */
+    if( status == ReportBuilderSuccess )
+    {
+        status = writeTaskIdsArray( pCurrentWritePos,
+                                    remainingBufferLength,
+                                    pMetrics->pTaskIdsArray,
+                                    pMetrics->taskIdsArrayLength,
+                                    &( bufferWritten ) );
+
+        if( status == ReportBuilderSuccess )
+        {
+            pCurrentWritePos += bufferWritten;
+            remainingBufferLength -= bufferWritten;
+        }
+        else
+        {
+            LogError( ( "Failed to write task ids array." ) );
```

Comment:
The error code can be logged as well
```suggestion
            LogError( ( "Failed to write custom metric list of task IDs to JSON report." ) );
```